internal/characters/kokomi/skill.go

```diff
@@ -81,22 +81,9 @@ func (c *char) skillTick(d *combat.AttackEvent) {
 		d.Info.FlatDmg = c.burstDmgBonus(d.Info.AttackTag)
 	}
 
-	maxhp := c.MaxHP()
-
 	c.Core.QueueAttackEvent(d, 0)
-	c.Core.Player.Heal(player.HealInfo{
-		Caller:  c.Index,
-		Target:  c.Core.Player.Active(),
-		Message: "Bake-Kurage",
-		Src:     skillHealPct[c.TalentLvlSkill()]*maxhp + skillHealFlat[c.TalentLvlSkill()],
-		Bonus:   d.Snapshot.Stats[attributes.Heal],
-	})
-
-	// Particles are 0~1 (1:2) on every damage instance
-	if c.Core.Rand.Float64() < .6667 {
-		c.Core.QueueParticle("kokomi", 1, attributes.Hydro, c.ParticleDelay)
-	}
 
+	maxhp := d.Snapshot.BaseHP*(1+d.Snapshot.Stats[attributes.HPP]) + d.Snapshot.Stats[attributes.HP]
```

Comment:
maybe merge healing from the skill and c2?
e.g.
```go
	maxhp := d.Snapshot.BaseHP*(1+d.Snapshot.Stats[attributes.HPP]) + d.Snapshot.Stats[attributes.HP]
	src := skillHealPct[c.TalentLvlSkill()]*maxhp + skillHealFlat[c.TalentLvlSkill()]

	// C2 handling - believe this is an additional instance of flat healing
	// Sangonomiya Kokomi gains the following Healing Bonuses with regard to characters with 50% or less HP via the following methods:
	// Kurage's Oath Bake-Kurage: 4.5% of Kokomi's Max HP.
	if c.Base.Cons >= 2 {
		active := c.Core.Player.ActiveChar()
		if active.HPCurrent/active.MaxHP() <= .5 {
			bonus := 0.045 * maxhp
			src += bonus
			c.Core.Log.NewEvent("kokomi c2 proc'd", glog.LogCharacterEvent, c.Index).
				Write("bonus", bonus)
		}
	}

	c.Core.Player.Heal(player.HealInfo{
		Caller:  c.Index,
		Target:  c.Core.Player.Active(),
		Message: "Bake-Kurage",
		Src:     src,
		Bonus:   d.Snapshot.Stats[attributes.Heal],
	})
```

Reply:
yes, do merge healing from skill and c2.